src/LibraryViewExtension/LibraryViewController.cs

```diff
@@ -217,6 +221,29 @@ private void OnBrowserConsoleMessage(object sender, ConsoleMessageEventArgs e)
             System.Diagnostics.Trace.Write(e.Message);
         }
 
+        internal static IDisposable SetupSearchModelEventsObserver(NodeSearchModel model, IEventController controller, int throttleTime = 200)
+        {
+            var observer = new EventObserver<NodeSearchElement, string>(
+                    nodes => controller.RaiseEvent("libraryDataUpdated", nodes),
+                    (s, e) => {
+                            var name = NodeItemDataProvider.GetFullyQualifiedName(e);
+                            return string.IsNullOrEmpty(s) ? name : string.Format("{0}, {1}", s, name);
+                        }
+                ).Throttle(TimeSpan.FromMilliseconds(throttleTime));
+
+            model.EntryAdded += observer.OnEvent;
+            model.EntryRemoved += observer.OnEvent;
+            model.EntryUpdated += observer.OnEvent;
+
+            return new AnonymousDisposable(() =>
+            {
+                model.EntryAdded -= observer.OnEvent;
+                model.EntryRemoved -= observer.OnEvent;
+                model.EntryUpdated -= observer.OnEvent;
+                observer.Dispose();
+            });
```

Comment:
I understand the intent here (and why it is required for the test case below), but it is in my opinion a little harder to understand with `AnonymousDisposable`. I might have attempted it slightly differently like so:

```csharp
var observer = new ...;

// Subscribe to a new callback even on observer
observer.OnDisposing += () => {
    // Unsubscribe from different events...
    model.EntryAdded -= observer.OnEvent;

    // At this point observer internally sets OnDisposing to null...
};

return observer;
```

Would that have caused any issue?